src/autolabel/transforms/custom_api.py

```diff
@@ -0,0 +1,136 @@
+
+            self.httpx = httpx
+            self.timeout_time = timeout
+            self.timeout = httpx.Timeout(connect=CONNECTION_TIMEOUT, timeout=timeout)
+            limits = httpx.Limits(
+                max_keepalive_connections=MAX_KEEPALIVE_CONNECTIONS,
+                max_connections=MAX_CONNECTIONS,
+                keepalive_expiry=timeout,
+            )
+            self.client = httpx.AsyncClient(
+                timeout=self.timeout, limits=limits, follow_redirects=True
+            )
+            self.client_with_no_verify = httpx.AsyncClient(
+                timeout=self.timeout, limits=limits, follow_redirects=True, verify=False
+            )
+        except ImportError:
+            raise ImportError(
+                "httpx and fake_useragent are required to use the custom API transform. Please install them with the following command: pip install httpx fake_useragent"
+            )
+
+    def name(self) -> str:
+        return TransformType.CUSTOM_API
+
+    async def _get_result(
+        self, url: str, verify=True, headers=HEADERS, retry_count=0
+    ) -> Dict[str, Any]:
+        if retry_count >= self.max_retries:
+            logger.warning(f"Max retries reached for URL: {url}")
+            raise TransformError(
+                TransformErrorType.MAX_RETRIES_REACHED,
+                f"Max retries reached for URL: {url}",
+            )
+
+        try:
+            client = self.client
+            if not verify:
+                client = self.client_with_no_verify
+            response = await client.get(url, headers=headers)
+            response.raise_for_status()
+            return response.text
+        except self.httpx.ConnectTimeout as e:
+            logger.error(f"Timeout when fetching content from URL: {url}")
+            raise TransformError(
+                TransformErrorType.TRANSFORM_TIMEOUT,
+                "Timeout when fetching content from URL",
+            )
+        except ssl.SSLCertVerificationError as e:
+            logger.warning(
+                f"SSL verification error when fetching content from URL: {url}, retrying with verify=False"
+            )
+            await asyncio.sleep(BACKOFF**retry_count)
+            return await self._get_result(
+                url, verify=False, headers=headers, retry_count=retry_count + 1
+            )
+        except Exception as e:
+            logger.error(f"Error fetching content from URL: {url}. Exception: {e}")
+            raise e
+
+    async def _apply(self, row: Dict[str, Any]) -> Dict[str, Any]:
+        for col in self.request_columns:
+            if col not in row:
+                logger.error(
+                    f"Missing request column: {col} in row {row}",
+                )
+        request_template = f"{self.base_url}?" + "&".join(
+            [f"{col}={{{col}}}" for col in self.request_columns if col in row]
+        )
+        url = request_template.format_map(defaultdict(str, row))
```

Comment:
let's use a standardized library like requests or similar to do this - we need to make sure parameter values are urlencoded etc.
```
r = requests.get(base_url, params={'key': 'value', ...})
print(r.url)
```